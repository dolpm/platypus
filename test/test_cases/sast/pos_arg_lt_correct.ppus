pipe test |> ['a, 'b, 'c] |> [x: &'a int, y: &'b int, z: &'c int] |> &'c int {
  if (@x == 2) {
    |> x;
  }
  if (@y == 2) {
    |> y;
  }
  |> z;
}

pipe main |> [] |> [] |> unit {
  int x <| 4;
  while (x < 5) {
    int y <| 4;
    loop 0 -> 5 as z {
      test <| [&x, &y, &z];
    }
  }
  |> ();
}
