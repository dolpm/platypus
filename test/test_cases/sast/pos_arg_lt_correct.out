pipe test |'a, 'b, 'c| [x: &'a int, y: &'b int, z: &'c int] |> &'c int {
{(id: test_with_args)
{(id: test_wrapper)
  if ((bool : (int : @(&'a int : x)) == (int : 2)))
  {(id: 2)
    |> (&'a int : x);
  }
  if ((bool : (int : @(&'b int : y)) == (int : 2)))
  {(id: 4)
    |> (&'b int : y);
  }
  |> (&'c int : z);
}
}
}

pipe main [] |> unit {
{(id: main_with_args)
{(id: main_wrapper)
  int x = (int : 4);
  while (bool : (int : x) < (int : 5))
  {(id: 6)
    int y = (int : 4);
    loop (int : 0) -> (int : 5) as (z,(int : 1))    {(id: 7)
      (&'_ int : test <| [(&'_ int : &(int : x)), (&'_ int : &(int : y)), (&'_ int : &(int : z))]);
    }
  }
  |> (unit : ());
}
}
}
